impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java

```diff
@@ -192,7 +192,7 @@ void doReturn(CodeAttribute b, ClassMethod method) {
                         if (Modifier.isFinal(method.getModifiers())) {
                             finalMethods.add(methodSignature);
                         }
-                        if (method.isBridge()) {
+                        if (method.isBridge() && !superClassAbstractAndPackagePrivate(cls)) {
```

Comment:
Hm, so any  bridge method declared on a subclass with package-private superclass will be omitted.. I don't think this is entirely correct. What if there is another bridge method that should be added, for example due to generics?